### PR TITLE
Byl8/challenges backend documentation

### PR DESCRIPTION
**Pull request by BrianLa0616**
### Summary 

Added comments for challenges model on flutter and removed dummy data.

### CODE REVIEW

**Comment by cathli66** [comment]
Thanks for adding more documentation for the backend methods! I was actually finished with connecting the Preview with the correct fields in my own PR... we should really communicate to each other when we're doing extra tasks, that's my bad!

**Comment by BrianLa0616** [comment]
> Thanks for adding more documentation for the backend methods! I was actually finished with connecting the Preview with the correct fields in my own PR... we should really communicate to each other when we're doing extra tasks, that's my bad!

**Comment by cathli66** [comment]
Another quick comment - we should use the LocationType enum more and store location as that type. Try to only convert between LocationType and string on the client facing ends (admin and game) rather than storing them as strings everywhere in the backend. Since our current codebase already has location as a string, we can treat this a task for later.

**Comment by BrianLa0616** [comment]
> Another quick comment - we should use the LocationType enum more and store location as that type. Try to only convert between LocationType and string on the client facing ends (admin and game) rather than storing them as strings everywhere in the backend. Since our current codebase already has location as a string, we can treat this a task for later.

I'm waiting for Nikita's PR on the filtering because he already has that implemented and we'll have to fix it anyways after he pushes it

**Comment by cathli66** [comment]
> > Another quick comment - we should use the LocationType enum more and store location as that type. Try to only convert between LocationType and string on the client facing ends (admin and game) rather than storing them as strings everywhere in the backend. Since our current codebase already has location as a string, we can treat this a task for later.
> 
> I'm waiting for Nikita's PR on the filtering because he already has that implemented and we'll have to fix it anyways after he pushes it

Just spoke to Nikita and he might redo his PR since you already added in a lot of the fields he needs. I still need to add in category into EventBase, so I can edit location to be LocationType in the same PR. Either way your current PR is fine :)

File: game/lib/challenges/challenges_page.dart [hunk 1]
```diff
@@ -136,18 +106,96 @@ class _ChallengesPageState extends State<ChallengesPage> {
                     ],
                   ),
                 ),
-                Expanded(
-                  child: ListView.separated(
-                    padding: const EdgeInsets.all(0),
-                    itemCount: cells.length,
+                Expanded(child: Consumer4<EventModel, GroupModel, TrackerModel,
```
Comment (neketka) [inline game/lib/challenges/challenges_page.dart:109]: It looks like this code is a repeat of what the master branch has. How is this different than the code we have currently?

File: game/lib/journeys/journeys_page.dart [hunk 2]
```diff
@@ -159,15 +175,15 @@ class _JourneysPageState extends State<JourneysPage> {
...
                                 Image.network(
                                     "https://picsum.photos/250?image=9"), // dummy data for now; should pass in thumbnail parameter
                                 event.description,
                                 locationCount,
                                 numberCompleted,
                                 complete,
                                 difficulty,
-                                event.minimumScore,
-                                0),
+                                total_points),
```
Comment (neketka) [inline game/lib/journeys/journeys_page.dart:186]: Nit: camelCase it

File: server/src/challenge/challenge.e2e-spec.ts [hunk 3]
```diff
@@ -167,7 +169,9 @@ describe('ChallengeModule E2E', () => {
       const secondChalDTO: ChallengeDto = {
```
Comment (neketka) [inline server/src/challenge/challenge.e2e-spec.ts:169]: Nit: Dto instead of DTO

File: server/src/challenge/challenge.e2e-spec.ts [hunk 4]
```diff
@@ -127,7 +127,9 @@ describe('ChallengeModule E2E', () => {
       const chaldto: ChallengeDto = {
```
Comment (neketka) [inline server/src/challenge/challenge.e2e-spec.ts:127]: Nit: chalDto

**Review by neketka — CHANGES_REQUESTED** [review]
Overall good, some minor nits

File: game/lib/challenges/challenges_page.dart (hunk 1 above)
Comment (BrianLa0616) [reply game/lib/challenges/challenges_page.dart:109]: It isn't. I just forgot to add documentation and I didn't create a new branch.... oops

**Review by cathli66 — APPROVED** [review]
Ok everything looks good! Thanks for fixing the difficulty tag on the frontend too

**Review by neketka — APPROVED** [review]
Addressed all my comments!

**Pull request merged**